bodhi/tests/server/services/test_updates.py

```diff
@@ -5106,3 +5107,122 @@ def test_unexpected_exception(self, log_exception, waive_test_results, *args):
+
+    @mock.patch.dict(config, [('greenwave_api_url', 'https://greenwave.api')])
+    @mock.patch('bodhi.server.util.call_api')
+    def test_get_test_results_erroring_on_greenwave(self, call_api, *args):
+        """
+        Ensure if all conditions are met we do try to call greenwave with the proper
+        argument but that call raises an error.
+        """
+        nvr = u'bodhi-2.0-1.fc17'
+        call_api.side_effect = requests.exceptions.HTTPError(
+            'Un-expected error foo bar')
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.locked = False
+
+        res = self.app.get('/updates/%s/get-test-results' % str(nvr), status=502)
+
+        call_api.assert_called_once_with(
+            'https://greenwave.api/decision',
+            data={
+                'product_version': u'fedora-17',
+                'decision_context': u'bodhi_update_push_testing',
+                'subject': [
+                    {'item': u'bodhi-2.0-1.fc17', 'type': 'koji_build'},
+                    {'original_spec_nvr': u'bodhi-2.0-1.fc17'},
+                    {'item': u'FEDORA-2018-a3bbe1a8f2', 'type': 'bodhi_update'}
+                ]
+            },
+            method='POST',
+            retries=3,
+            service_name='Greenwave'
+        )
+
+        self.assertEquals(
+            res.json_body,
+            {
+                u'errors': [
+                    {
+                        u'description': u'Un-expected error foo bar',
+                        u'location': u'body',
+                        u'name': u'request'
+                    }
+                ],
+                u'status': u'error'}
+        )
+
+    @mock.patch.dict(config, [('greenwave_api_url', 'https://greenwave.api')])
+    @mock.patch('bodhi.server.util.call_api')
+    def test_get_test_results_calling_greenwave(self, call_api, *args):
+        """
+        Ensure if all conditions are met we do try to call greenwave with the proper
+        argument.
+        """
+        nvr = u'bodhi-2.0-1.fc17'
+        call_api.return_value = {"foo": "bar"}
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.locked = False
+
+        res = self.app.get('/updates/%s/get-test-results' % str(nvr))
+
+        call_api.assert_called_once_with(
+            'https://greenwave.api/decision',
+            data={
+                'product_version': u'fedora-17',
+                'decision_context': u'bodhi_update_push_testing',
+                'subject': [
+                    {'item': u'bodhi-2.0-1.fc17', 'type': 'koji_build'},
+                    {'original_spec_nvr': u'bodhi-2.0-1.fc17'},
+                    {'item': u'FEDORA-2018-a3bbe1a8f2', 'type': 'bodhi_update'}
+                ]
+            },
+            method='POST',
+            retries=3,
+            service_name='Greenwave'
+        )
+
+        self.assertEquals(res.json_body, {u'decision': {u'foo': u'bar'}})
```

Comment:
Let's add one more test for a 504 gateway timeout. This will help us differentiate between problems if Bodhi can't communicate with Greenwave.